src/commands/deploy.ts

```diff
@@ -105,13 +105,11 @@ export async function deploy(context: IDeployWizardContext, confirmDeployment: b
         await javaUtils.configureJavaSEAppSettings(node);
     }
 
-    workspaceConfig = workspace.getConfiguration(constants.extensionPrefix, Uri.file(context.fsPath));
-
     const currentWorkspace: WorkspaceFolder | undefined = workspaceUtil.getContainingWorkspace(context.fsPath);
     if (currentWorkspace && (isPathEqual(currentWorkspace.uri.fsPath, context.fsPath) || isSubpath(currentWorkspace.uri.fsPath, context.fsPath))) {
         // currentWorkspace is only set if there is one active workspace
         // only check enableScmDoBuildDuringDeploy if currentWorkspace matches the workspace being deployed as a user can "Browse" to a different project
-        if (workspaceConfig.get(constants.configurationSettings.showBuildDuringDeployPrompt)) {
+        if (getWorkspaceSetting<boolean>(constants.configurationSettings.showBuildDuringDeployPrompt, currentWorkspace.uri.fsPath)) {
```

Comment:
Here you use `currentWorkspace.uri.fsPath`, but below you use `context.fsPath`. No strong preference on which one you use, but I'd prefer to be consistent.